### PR TITLE
Minor log optimization for easily investigating mastership handoff latency

### PR DESCRIPTION
**Pull request by i3wangyi**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

#575 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

(Write a concise description including what, why, how)

Make the log for visible and human-readable when any mastership handoff happens

### Tests

- [x] The following tests are written for this issue:

(List the names of added unit/integration tests)

- [x] The following is the result of the "mvn test" command on the appropriate module:

(Copy & paste the result of "mvn test")

[ERROR] Tests run: 882, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 3,710.718 s <<< FAILURE! - in TestSuite
[ERROR] testTaskPerformanceMetrics(org.apache.helix.monitoring.mbeans.TestTaskPerformanceMetrics)  Time elapsed: 2.329 s  <<< FAILURE!
java.lang.AssertionError: expected:<true> but was:<false>
        at org.apache.helix.monitoring.mbeans.TestTaskPerformanceMetrics.testTaskPerformanceMetrics(TestTaskPerformanceMetrics.java:118)

[ERROR] testForceDeleteJobFromJobQueue(org.apache.helix.integration.task.TestDeleteJobFromJobQueue)  Time elapsed: 0.475 s  <<< FAILURE!
org.apache.helix.HelixException: Failed to delete job: testForceDeleteJobFromJobQueue_job2 from queue: testForceDeleteJobFromJobQueue
        at org.apache.helix.integration.task.TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue(TestDeleteJobFromJobQueue.java:75)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue:75 » Helix Failed to ...
[ERROR]   TestTaskPerformanceMetrics.testTaskPerformanceMetrics:118 expected:<true> but was:<false>
[INFO] 
[ERROR] Tests run: 882, Failures: 2, Errors: 0, Skipped: 0
[INFO] 

The failed 2 tests succeed when running in IDE

### Commits

- [x] My commits all reference appropriate Apache Helix GitHub issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Code Quality

- [x] My diff has been formatted using helix-style.xml

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/manager/zk/DistributedLeaderElection.java [hunk 1]
```diff
@@ -87,36 +88,39 @@ public synchronized void onControllerChange(NotificationContext changeContext) {
...
       ControllerManagerHelper controllerHelper) {
-    LOG.info(manager.getInstanceName() + " relinquish leadership for cluster: " + manager
+    long start = System.currentTimeMillis();
+    LOG.info(manager.getInstanceName() + " tries to relinquish leadership for cluster: " + manager
         .getClusterName());
     controllerHelper.stopControllerTimerTasks();
     controllerHelper.removeListenersFromController(_controller);
     // clear write-through cache
     manager.getHelixDataAccessor().getBaseDataAccessor().reset();
+    LOG.info("{} relinquishes leadership for cluster: {}, took: {}", manager.getInstanceName(),
```
Comment (jiajunwang) [inline helix-core/src/main/java/org/apache/helix/manager/zk/DistributedLeaderElection.java]: Correct me if I am wrong, I think we record every "onControllerChange" processing time. So it should be easy to derive the latency from that log. I feel this log would be redundant a little bit.

Comment (i3wangyi) [reply helix-core/src/main/java/org/apache/helix/manager/zk/DistributedLeaderElection.java]: It just makes the debugging process a little easier. Plus, it only adds the time spent on the method only in addition to the existing log. The "onControllerChange" process time can be the additional weapon to figure out the remaining latency other than the method.

**Comment by i3wangyi** [comment]
> I don't have a strong preference here. But we tend to be conservative on adding new logs. This has been complained about by our customers before.
> So,
> 
> 1. If this is for debug, will log.debug be enough?
> 2. If the concerning part is listener reset, shall we directly log the latency of the reset inside that method?

I added 1 line at the (leader -> standby) relinquish stage.
The other log changes are just either 1. move the log position 2. add the time duration. I don't think overall it will make the current log situation worse. 

The purpose of the changes will help us quickly figure out the latency issue while searching the logs; Compared with the current approach: grep through logs and connecting dots here and there. Right?

Too many logs is a problem in Helix controller; As a side project to this, I hope I will have some time to analyze the current logs and find things that can be discarded.

**Comment by i3wangyi** [comment]
The PR is ready to merge into master, approved by @jiajunwang

**Pull request merged**